File: 2015/go/src/day3A.go
package main

import (
	"fmt"
	"strconv"
)

func followDirections(directions string) map[string]int {
	path := map[string]int{
		"0,0": 1,
	}

	pos := [2]int{0, 0}

	for _, dir := range directions {
		if dir == '^' {
			pos[1]++
		} else if dir == 'v' {
			pos[1]--
		} else if dir == '<' {
			pos[0]--
		} else if dir == '>' {
			pos[0]++
		}
		charPos := strconv.Itoa(pos[0]) + "," + strconv.Itoa(pos[1])
		path[charPos]++
	}

	return path
}

func main() {
	input := readFile("day3.dat")

	fmt.Println(len(followDirections(input)))
}

File: 2020/python/day3A.py
import common

def main():
    dat = common.readFile('day3.dat')
    
    print(toboggan(dat))

def toboggan(dat):
    pos = 0
    tree_count = 0
    i = 1

    while i < len(dat):
        row = list(dat[i])
        
        pos = (pos + 3) % len(row)
        
        if row[pos] == '#':
            tree_count = tree_count + 1

        i = i + 1

    return tree_count


if __name__ == '__main__':
    main()